### PR TITLE
feat(XYZ Source): Extending the XYZ source interface

### LINKED ISSUE #43: Extend the XYZ Source interface
Need to include the initialization options for XYZ as props on the XYZSource component.

### PR DESCRIPTION
**Pull request by nickmanks**
Users can now pass initialization options for the OpenLayers XYZ source class via props on the OLR XYZSource component:

```html
<XYZSource cacheSize={1024} url={'some-url.com'} ...\>
``` 

All the OpenLayers XYZ init options are mapped to the components props.

Closes #43

### CODE REVIEW

File: src/source/xyz.js [hunk 1]
```diff
@@ -10,32 +10,45 @@ import {SourceCtx} from '.';
...
+    const {props} = this;
+    const {layer} = props;
 
     this.source = new XYZ({
-      tileSize,
-      tileGrid,
-      projection,
-      url,
-      urls
+      ...props, layer: undefined, refreshKey: undefined
```
Comment (kollhof) [inline src/source/xyz.js]: Lets be more explicit about what we pass to the constructor and only pass what we need to.

Comment (kollhof) [reply src/source/xyz.js]: This pattern may appear to work but it not what you probably intended.

Run the following:
```javascript
console.log({...{foo:123, bar: 456}, foo: undefined})
```

You can see that the object still contains a `foo` prop.
So, setting options props that are not accepted by the constructor to undefined is not quite the same as calling the constructor with only accepted option props.

File: src/source/xyz.test.js [hunk 2]
```diff
@@ -85,15 +117,90 @@ describe('<XYZSource />', ()=> {
...
+  });
+});
+
+
+describe('<XYZSource /> - refresh key', ()=> {
+  it('can be refreshed by changing the refresh key', ()=> {
+    const layer = new OlTileLayer();
+    const projection = getProjection('EPSG:3857');
+    const rendered = renderer.create(
+      <XYZSource layer={layer} projection={projection} refreshKey={undefined} />
```
[the remark this answers is not in the document]
Comment (nickmanks) [reply src/source/xyz.test.js]: Just for explicit purposes in testing?

File: src/source/xyz.js (hunk 1 above)
Comment (nickmanks) [reply src/source/xyz.js]: Good point. Will update.

**Pull request merged**